### PR TITLE
allow to disable http

### PR DESCRIPTION
**Pull request by vladak**
This change allows to build without HTTP support in various places. Specifically, this is handy for Solaris builds of the wanboot loader, however can be useful for general minimization.

Using the `no-http` configure option automatically disables OCSP support.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
This contribution is well beyond what we'll accept with a [CLA](https://www.openssl.org/policies/cla.html).  Could you fill in an [ICLA](https://www.openssl.org/policies/openssl_icla.pdf) please?  If this contribution is part of your employment, we'll also need a CCLA.

**Comment by vladak** [comment]
> This contribution is well beyond what we'll accept with a [CLA](https://www.openssl.org/policies/cla.html). Could you fill in an [ICLA](https://www.openssl.org/policies/openssl_icla.pdf) please? If this contribution is part of your employment, we'll also need a CCLA.

Sure, ICLA sent. This is part of the work I do for Oracle, which has CCLA, I believe.

File: .github/workflows/run-checker-ci.yml [hunk 1]
```diff
@@ -23,6 +23,7 @@ jobs:
           no-dtls,
           no-ec,
           no-ec2m,
+          no-http,
           no-siv,
           no-legacy,
           no-rfc3779,
```
Comment (t8m) [inline .github/workflows/run-checker-ci.yml:29]: To compensate for this added build, please remove the no-rfc3779 build - it is already in the run-checker-daily run anyway.

**Review by t8m — CHANGES_REQUESTED** [review]

File: .github/workflows/run-checker-ci.yml (hunk 1 above)
Comment (vladak) [reply .github/workflows/run-checker-ci.yml]: okay, done.

File: Configure [hunk 2]
```diff
@@ -1779,6 +1780,8 @@ unless ($disabled{winstore}) {
     }
 }
 
+disable('no-http-api', 'ocsp') if ($disabled{http});
+
```
[the remark this answers is not in the document]
Comment (vladak) [reply Configure]: I see, fixed.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
@paulidale please reconfirm

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged, thanks for the contribution.

**Pull request closed without merging**